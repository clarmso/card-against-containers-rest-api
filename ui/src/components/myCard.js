import PropTypes from "prop-types"
import React from "react"

import { makeStyles } from "@material-ui/core/styles"
import { Card, CardActions, CardContent } from "@material-ui/core"
import ComputerIcon from "@material-ui/icons/Computer"
import CircularProgress from '@material-ui/core/CircularProgress';

const CARD_WIDTH = "250px"
const CARD_HEIGHT = "250px"
const BLACK = "#322923"

const useStylesQuestion = makeStyles({
  card: {
    backgroundColor: BLACK,
    color: "white",
    width: CARD_WIDTH
  },
  content: {
    height: CARD_HEIGHT
  }
})

const useStylesAnswer = makeStyles({
  card: {
    backgroundColor: "white",
    color: BLACK,
    width: CARD_WIDTH
  },
  content: {
    height: CARD_HEIGHT
  }
})

const myCard = (content, classes) => {
  return (
    <Card raised={true} className={classes.card}>
      <CardContent className={classes.content}>
        {!content ? <CircularProgress size={20} /> : content}
      </CardContent>
      <CardActions>
        <ComputerIcon />
        <span>Cards Against Containers</span>
      </CardActions>
    </Card>
  )
}

const Question = ({ question }) => {
  return myCard(question, useStylesQuestion());
}

const Answer = ({answer}) => {
  return myCard(answer, useStylesAnswer())
}

Question.propTypes = {
  question: PropTypes.string,
}

Question.defaultProps = {
  question: "",
}

Answer.propTypes = {
  answer: PropTypes.string,
}

Answer.defaultProps = {
  answer: "",
}

export {Question, Answer}