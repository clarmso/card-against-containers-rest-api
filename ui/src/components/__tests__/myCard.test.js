import React from "react"
import renderer from "react-test-renderer"
import { Question, Answer } from "../myCard"

describe("Cards", () => {
  describe("<Question />", () => {
    test("Display a question card without any content", () => {
      const component = renderer.create(<Question />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    test("Display a question card", () => {
      const component = renderer.create(
        <Question question="What is Stratford ON?" />
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe("<Answer />", () => {
    test("Display an answer card without any content", () => {
      const component = renderer.create(<Answer i />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })

    test("Display an answer card", () => {
      const component = renderer.create(
        <Answer answer="The swan is a symbol of this Ontario city; Each year, white & black swans are released into the Avon River." />
      )
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
