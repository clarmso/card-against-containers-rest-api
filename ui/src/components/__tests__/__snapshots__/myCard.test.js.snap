// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards <Answer /> Display an answer card 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation8 MuiPaper-rounded"
  data-cy="answer-TheswanisasymbolofthisOntariocityEachyearwhiteblackswansarereleasedintotheAvonRiver"
>
  <div
    className="MuiCardContent-root makeStyles-content-4"
  >
    The swan is a symbol of this Ontario city; Each year, white & black swans are released into the Avon River.
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 18c1.1 0 1.99-.9 1.99-2L22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z"
      />
    </svg>
    <span>
      Cards Against Containers
    </span>
  </div>
</div>
`;

exports[`Cards <Answer /> Display an answer card without any content 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-elevation8 MuiPaper-rounded"
  data-cy="answer-"
>
  <div
    className="MuiCardContent-root makeStyles-content-4"
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 18c1.1 0 1.99-.9 1.99-2L22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z"
      />
    </svg>
    <span>
      Cards Against Containers
    </span>
  </div>
</div>
`;

exports[`Cards <Question /> Display a question card 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation8 MuiPaper-rounded"
  data-cy="question-WhatisStratfordON"
>
  <div
    className="MuiCardContent-root makeStyles-content-2"
  >
    What is Stratford ON?
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 18c1.1 0 1.99-.9 1.99-2L22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z"
      />
    </svg>
    <span>
      Cards Against Containers
    </span>
  </div>
</div>
`;

exports[`Cards <Question /> Display a question card without any content 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-card-1 MuiPaper-elevation8 MuiPaper-rounded"
  data-cy="question-"
>
  <div
    className="MuiCardContent-root makeStyles-content-2"
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 18c1.1 0 1.99-.9 1.99-2L22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2H0v2h24v-2h-4zM4 6h16v10H4V6z"
      />
    </svg>
    <span>
      Cards Against Containers
    </span>
  </div>
</div>
`;
